### ISSUE TITLE
Creating Segment at the Action Level

### ISSUE THREAD

**Issue by Chardonneaur** [piwik/piwik#22712]
Hi Team,

Currently segments are only applicable on sessions only.
Piano Analytics, a competitor of Matomo is offering two modes: visits, actions.
From what I remembered, in the past, Matomo was action based regarding segments.

--> Would it be possible to offer the two modes?

**Comment by sgiehl** [comment]
@Chardonneaur Wouldn't the `action` mode be more like what a filter as described here https://github.com/matomo-org/matomo/issues/15438 would do?

**Comment by Chardonneaur** [comment]
@sgiehl I think so, is this feature [General]enable_segments_cache=0 available in the cloud?

**Comment by sgiehl** [comment]
I don't think so. It also disables the segment cache which slows down archiving

**Comment by Chardonneaur** [comment]
Another end user asked it today.

**Comment by Chardonneaur** [comment]
I released a plugin for this: https://github.com/Chardonneaur/ActionFilter

**Comment by Chardonneaur** [comment]
I released a plugin for this: https://github.com/Chardonneaur/ActionFilter
